source/common/router/config_impl.h

```diff
@@ -457,6 +457,56 @@ class RouteTracingImpl : public RouteTracing {
   Tracing::CustomTagMap custom_tags_;
 };
 
+/**
+ * Implementation of PathMatchPolicy that reads from the proto
+ * PathMatchPolicy of the RouteAction.
+ */
+class PathMatchPolicyImpl : public PathMatchPolicy {
+public:
+  PathMatchPolicyImpl(const envoy::config::core::v3::TypedExtensionConfig typed_config,
+                      ProtobufMessage::ValidationVisitor& validator, std::string path);
+
+  // Default constructor that disables internal redirect.
+  PathMatchPolicyImpl();
+
+  bool enabled() const override { return enabled_; }
+
+  PathMatchPredicateSharedPtr predicate() const override;
+
+  std::string path() const { return path_; }
+
+private:
+  const std::string path_;
+  const bool enabled_;
+  ProtobufTypes::MessagePtr predicate_config_;
+  PathMatchPredicateSharedPtr predicate_;
+};
+
+/**
+ * Implementation of PathRewritePolicyImpl that reads from the proto
+ * PathRewritePolicyImpl of the RouteAction.
+ */
+class PathRewritePolicyImpl : public PathRewritePolicy {
+public:
+  PathRewritePolicyImpl(const envoy::config::core::v3::TypedExtensionConfig typed_config,
+                        ProtobufMessage::ValidationVisitor& validator, std::string route_url);
+
+  // Default constructor that disables internal redirect.
+  PathRewritePolicyImpl();
+
+  bool enabled() const override { return enabled_; }
+
+  PathRewritePredicateSharedPtr predicate() const override;
+
+  std::string routeUrl() const { return route_url_; }
```

Comment:
Same here, unused method that's the only use of the field. Also, the value passed in to initialize is the route name, not a URL.

Reply:
Removed it.